### PR TITLE
feat: Gidk-2k9fpj | Shalinee-Prateek | Prod | Create a separate sidebar on GST module

### PR DESCRIPTION
**Pull request by shalinee-walkover**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
https://app.clickup.com/t/2k9fpj

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

File: apps/web-giddh/src/app/gst/filing/filing.component.ts [hunk 1]
```diff
@@ -160,4 +175,79 @@ export class FilingComponent implements OnInit, OnDestroy {
...
         this.selectTab('', false, '1. Overview');
     }
+
+    /**
+     * Navigates to the overview or dashboard page
+     *
+     * @param {*} type Type of report (gstr1, gstr2, gstr3b)
+     * @memberof FilingComponent
+     */
+    public navigateToOverview(type) {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/gst/filing/filing.component.ts:185]: return type

File: apps/web-giddh/src/app/gst/filing/filing.component.ts [hunk 2]
```diff
@@ -160,4 +175,79 @@ export class FilingComponent implements OnInit, OnDestroy {
...
+        this._route.navigate(['pages', 'gstfiling', 'filing-return'], { queryParams: { return_type: type, from: this.currentPeriod.from, to: this.currentPeriod.to, tab: 0, selectedGst: this.activeCompanyGstNumber } });
+    }
+
+    /**
+     * Navigates to GSTR 3B
+     *
+     * @param {*} type Type of report (gstr1, gstr2, gstr3b)
+     * @memberof FilingComponent
+     */
+    public navigateTogstR3B(type) {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/gst/filing/filing.component.ts:195]: param and return type missing

**Review by aditi-tiwari17 — APPROVED** [review]

File: apps/web-giddh/src/app/gst/gst-sidebar/gst-sidebar.component.scss [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+                ul {
+                    li {
+                        padding: 0px;
+                        a {
+                            display: flex;
+                            align-items: flex-start;
+                            padding: 7px 15px;
+                            span {
+                                i {
+                                    color: grey;
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/gst/gst-sidebar/gst-sidebar.component.scss:33]: use color code

File: apps/web-giddh/src/app/gst/gst-sidebar/gst-sidebar.component.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+            this.router.navigate(['/pages/home']);
+        }
+    }
+
+    /**
+    * This is navigate menu item
+    *
+    * @memberof GstrSidebarComponent
+    */
+    public navigate(type) {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/gst/gst-sidebar/gst-sidebar.component.ts:61]: param and return type missing

File: apps/web-giddh/src/app/gst/gstR1/gstR1.component.html [hunk 5]
```diff
@@ -1,3 +1,6 @@
+<div class="help-support-aside setting-sidebar-wrapper" *ngIf="asideGstSidebarMenuState === 'in'">
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/gst/gstR1/gstR1.component.html:1]: what's the use of blank div?

File: apps/web-giddh/src/app/gst/gstR1/gstR1.component.ts [hunk 6]
```diff
@@ -78,5 +111,12 @@ export class FileGstR1Component implements OnInit {
         });
 
     }
-
+    /**
+    * Unsubscribes from subscription
+    *
+    * @memberof FileGstR1Component
+    */
+    public ngOnDestroy(){
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/gst/gstR1/gstR1.component.ts:119]: return type

File: apps/web-giddh/src/app/gst/gstR2/gstR2.component.ts [hunk 7]
```diff
@@ -107,4 +140,12 @@ export class FileGstR2Component implements OnInit {
 
         });
     }
+    /**
+     * Unsubscribes from subscription
+     *
+     * @memberof FileGstR2Component
+     */
+    public ngOnDestroy(){
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/gst/gstR2/gstR2.component.ts:148]: return type

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/gst/gstR1/gstR1.component.html (hunk 5 above)
Comment (prateek-giddh) [reply apps/web-giddh/src/app/gst/gstR1/gstR1.component.html:1]: For backdrop of sidemenu

File: apps/web-giddh/src/app/gst/filing/filing.component.ts (hunk 1 above)
Comment (prateek-giddh) [reply apps/web-giddh/src/app/gst/filing/filing.component.ts:185]: Done

File: apps/web-giddh/src/app/gst/filing/filing.component.ts (hunk 2 above)
Comment (prateek-giddh) [reply apps/web-giddh/src/app/gst/filing/filing.component.ts:195]: Done

File: apps/web-giddh/src/app/gst/gst-sidebar/gst-sidebar.component.scss (hunk 3 above)
Comment (prateek-giddh) [reply apps/web-giddh/src/app/gst/gst-sidebar/gst-sidebar.component.scss:33]: Done

File: apps/web-giddh/src/app/gst/gst-sidebar/gst-sidebar.component.ts (hunk 4 above)
Comment (prateek-giddh) [reply apps/web-giddh/src/app/gst/gst-sidebar/gst-sidebar.component.ts:61]: Done

File: apps/web-giddh/src/app/gst/gstR1/gstR1.component.ts (hunk 6 above)
Comment (prateek-giddh) [reply apps/web-giddh/src/app/gst/gstR1/gstR1.component.ts:119]: Done

File: apps/web-giddh/src/app/gst/gstR2/gstR2.component.ts (hunk 7 above)
Comment (prateek-giddh) [reply apps/web-giddh/src/app/gst/gstR2/gstR2.component.ts:148]: Done

**Review by ravinderwalkover — APPROVED** [review]

**Pull request merged**